### PR TITLE
privilege: fix compatibility of DBIsVisible

### PR DESCRIPTION
**Pull request by imtbkcat**
### What problem does this PR solve? 
User with `trigger/references/execute` global privileges can not see all database names and use them, but in MySQL is okay.

in TiDB:
```
mysql u1@127.0.0.1:(none)> show grants;
+-----------------------------------------------------+
| Grants for User                                     |
+-----------------------------------------------------+
| GRANT References,Execute,Trigger ON *.* TO 'u1'@'%' |
+-----------------------------------------------------+
1 row in set
Time: 0.014s
mysql u1@127.0.0.1:(none)> show databases;
+--------------------+
| Database           |
+--------------------+
| INFORMATION_SCHEMA |
+--------------------+
1 row in set
```

in MySQL:
```
mysql u1@127.0.0.1:test> show grants;
+----------------------------------+
| Grants for u1@%                  |
+----------------------------------+
| GRANT TRIGGER ON *.* TO 'u1'@'%' |
+----------------------------------+
1 row in set
Time: 0.014s
mysql u1@127.0.0.1:test> show databases;
+--------------------+
| Database           |
+--------------------+
| information_schema |
| mysql              |
| performance_schema |
| sys                |
| test               |
+--------------------+
5 rows in set
Time: 0.014s
mysql u1@127.0.0.1:test> use test;
You are now connected to database "test" as user "u1"
Time: 0.001s
mysql u1@127.0.0.1:test> show tables;
+----------------+
| Tables_in_test |
+----------------+
| Ata            |
| atA            |
+----------------+
2 rows in set
Time: 0.014s
mysql u1@127.0.0.1:test>

mysql u1@127.0.0.1:test> show grants;
+----------------------------------+
| Grants for u1@%                  |
+----------------------------------+
| GRANT EXECUTE ON *.* TO 'u1'@'%' |
+----------------------------------+
1 row in set
Time: 0.015s
mysql u1@127.0.0.1:test> show databases;
+--------------------+
| Database           |
+--------------------+
| information_schema |
| mysql              |
| performance_schema |
| sys                |
| test               |
+--------------------+
5 rows in set

```
### What is changed and how it works?
add `TriggerPriv`, `EcecutePriv`, `ReferencePriv` into `globalDBVisible `

### Check List 

Tests 

 - Unit test

Code changes

 - None

Side effects

 - None

Related changes

 - Need to cherry-pick to the release branch

Release note

### CODE REVIEW

File: privilege/privileges/cache_test.go [hunk 1]
```diff
@@ -553,4 +553,25 @@ func (s *testCacheSuite) TestDBIsVisible(c *C) {
...
+	isVisible = p.DBIsVisible("testvisdb6", "%", "visdb")
+	c.Assert(isVisible, IsTrue)
+	mustExec(c, se, "TRUNCATE TABLE mysql.user")
+
+	mustExec(c, se, `INSERT INTO mysql.user (Host, User, Execute_priv) VALUES ("%", "testvisdb6", "Y")`)
+	err = p.LoadUserTable(se)
+	c.Assert(err, IsNil)
+	isVisible = p.DBIsVisible("testvisdb6", "%", "visdb")
+	c.Assert(isVisible, IsTrue)
+	mustExec(c, se, "TRUNCATE TABLE mysql.user")
```
Comment (crazycs520) [inline privilege/privileges/cache_test.go]: The added tests can pass in the master too?

**Comment by tiancaiamao** [comment]
LGTM

File: privilege/privileges/cache_test.go (hunk 1 above)
Comment (imtbkcat) [reply privilege/privileges/cache_test.go]: fixed

**Comment by tiancaiamao** [comment]
Ping @crazycs520

**Pull request merged**